### PR TITLE
Dynamic `offers_*` tables interaction class complete.

### PR DESCRIPTION
**Pull request by sammdu**
#29 

Also modified `Conn` to accept database name as a parameter.

Yet to complete testing.

### CODE REVIEW

File: src/main/java/tech/autodirect/api/database/TableOffers.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        stmt.close();
+    }
+
+    public void removeAllOffers() throws SQLException {
+        Statement stmt = this.db_conn.createStatement();
+        stmt.executeUpdate("DELETE FROM " + this.table_name + ";");
+        stmt.close();
+    }
+
+    private static HashMap offerResultToHashMap(ResultSet rs) throws SQLException {
```
Comment (Shalev-Lifshitz) [inline src/main/java/tech/autodirect/api/database/TableOffers.java:127]: Raw use of HashMap, consider replacing with HashMap<String, Object> as its better for documentation purposes

File: src/main/java/tech/autodirect/api/database/TableOffers.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        rs.next();
+        HashMap<String, Object> offer_result = offerResultToHashMap(rs);
+        stmt.close();
+
+        return offer_result;
+    }
+
+    public ArrayList getAllOffers() throws SQLException {
+        // construct a prepared SQL statement selecting all offers
+        Statement stmt = this.db_conn.createStatement();
```
Comment (Shalev-Lifshitz) [inline src/main/java/tech/autodirect/api/database/TableOffers.java:160]: This is not a prepared statement, right?

Comment (sammdu) [reply src/main/java/tech/autodirect/api/database/TableOffers.java:160]: Ah cuz here we are not inserting values so there is no injection risk.
The `table_name` cannot be compiled as part of prepared statements afaik.

Comment (Shalev-Lifshitz) [reply src/main/java/tech/autodirect/api/database/TableOffers.java:237]: @sammdu Right, I mean that the comment says it's a prepared statement when it's not.

Comment (Shalev-Lifshitz) [reply src/main/java/tech/autodirect/api/database/TableOffers.java:237]: > Ah cuz here we are not inserting values so there is no injection risk. The `table_name` cannot be compiled as part of prepared statements afaik.

As a side note, I'm pretty sure you can still use a PreparedStatement by just passing "SELECT * FROM " + this.table_name + ";" to the constructor. Cause it's just a string.

**Review by Shalev-Lifshitz — APPROVED** [review]
Great job, I enjoyed reading through it and seeing your thought process :-D

**Pull request merged**